carbon-design-system/carbon-addons-iot-react
# feat(table): enable single nested hierarchy styling

Closes #1602
Closes #1600 

**Summary**

- The current Table only supports multiple hierarchy nested styling, but the release of List has given a new use-case where single nested hierarchies can exist, and should have similar styling to List
- This PR adds a new prop `hasSingleNestedHierarchy` as an option that will disable the gray offset if enabled

Related internal Monitor design: https://ibm.invisionapp.com/share/KWO0FZ5BHQ9#/screens/319695909

**Change List (commits, features, bugs, etc)**

- added `hasSingleNestedHierarchy` prop
- removed padding when select all checkbox was enabled as it was creating an odd offset where it was not aligned with the first level parents (this also fixed #1600 )
- Updated snapshots, but not sure why the pagination id increased
- Added new single nested hierarchy story

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/watson-iot-table--stateful-example-with-single-nested-hierarchy in the netlify preview
  - ensure that all first level checkboxes are aligned
  - expand one of the rows
    - see that there is no gray offset

**Regression Test (how to make sure I didn't break things)**
- Go to ?path=/story/watson-iot-table--stateful-example-with-row-nesting-and-fixed-columns
  - ensure that the checkboxes are still aligned
  - expand one of the rows
    - see that there IS a gray offset still (as this is not a single hierarchy)

- Go to table resize AND fixed column stories to ensure that the padding still looks correct in the select all checkbox in the table head AND the offset still works correctly with those (if they include nested rows)


![Screen Shot 2020-09-16 at 12 40 17 PM](https://user-images.githubusercontent.com/25177649/93374234-08490f80-f81c-11ea-8c7d-3d7f90119f30.png)


![Screen Shot 2020-09-16 at 12 40 13 PM](https://user-images.githubusercontent.com/25177649/93374246-0da65a00-f81c-11ea-8710-ec913d14bc19.png)
